python/dpu_utils/ptutils/componenttrainer.py

```diff
@@ -105,6 +107,15 @@ def validation_tensors():
                 max_queue_size=10 * self.__minibatch_size
             )
 
+        def minibatch_iterator(data_iterator):
+            while True:
+                mb_data, data_iterator_exhausted, num_elements = self.__model.create_minibatch(data_iterator,
+                                                                                               max_num_items=self.__minibatch_size)
+                yield mb_data, data_iterator_exhausted, num_elements
+
+                if data_iterator_exhausted:

```

Comment:
I'm a bit concerned about the semantics here - what happens if `create_minibatch` provides an empty `mb_data` (because the last minibatch used up all available data)? Would it make more sense to drop the `data_iterator_exhausted` return bit and simply use

```
            while True:
                mb_data, num_elements = self.__model.create_minibatch(
                    data_iterator,
                    max_num_items=self.__minibatch_size,
                )
                if num_elements == 0:
                    break
                yield mb_data, num_elements
```
The check on not using partial batches could then be moved to `create_minibatch` as well (i.e., if you would have partial batch, just return `num_elements = 0`) - at the moment this seems to occur in a number of places.